### PR TITLE
Update eslint config and eslint to v6

### PR DESCRIPTION
**Pull request by nathanmarks**
Updated eslint and the config, and fixed some lint errors that the newer eslint version exposed.

### CODE REVIEW

File: src/styles/SetStyles.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import Radium, { Style } from 'radium'
-import { Normalize, GlobalTheme } from '../index'
+import Normalize from './Normalize'
```
Comment (nathanmarks) [inline src/styles/SetStyles.js]: circular dep

File: src/components/Typography/docs/helpers.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable react/jsx-no-undef */
+/* eslint-disable react/jsx-no-undef, react/prop-types */
```
Comment (nathanmarks) [inline src/components/Typography/docs/helpers.js]: there's a component here with no prop type defs

File: src/utils/detectFeature.js [hunk 3]
```diff
@@ -1,5 +1,12 @@
 import memoize from './memoize'
 
+export const isNodeEnv = () => typeof window === 'undefined'
```
Comment (nathanmarks) [inline src/utils/detectFeature.js]: no-use-before-define

File: src/components/ScrollTrack/ScrollTrack.js [hunk 4]
```diff
@@ -238,9 +238,9 @@ class ScrollTrack extends Component {
     } // already sliding
 
     const { parentWidth, trackWidth } = this.getNodeWidths()
+    const { onBeforeNext, scrollOffset } = this.props
```
Comment (nathanmarks) [inline src/components/ScrollTrack/ScrollTrack.js]: no-use-before-define

File: src/styles/SetStyles.js (hunk 1 above)
Comment (dcocchia) [reply src/styles/SetStyles.js]: oo tricky one

**Pull request merged**